automation1App/src/Automation1MotorController.cpp

```diff
@@ -606,34 +588,100 @@ asynStatus Automation1MotorController::readbackProfile()
         pAxes_[i]->readbackProfile();
     }
 
-    // Clear readback command.  This is a "busy" record, don't want to do this until readback is complete.
     readbackStatus = readbackOK ? PROFILE_STATUS_SUCCESS : PROFILE_STATUS_FAILURE;
     setIntegerParam(profileReadbackStatus_, readbackStatus);
+
+    // Clear readback command.  This is a "busy" record, don't want to do this until readback is complete.
     setIntegerParam(profileReadback_, 0);
     setIntegerParam(profileReadbackState_, PROFILE_READBACK_DONE);
     callParamCallbacks();
 
     return readbackOK ? asynSuccess : asynError;
 }
 
+asynStatus Automation1MotorController::poll()
+{
+    Automation1DataCollectionStatus dataCollectionStatus;
+    Automation1TaskStatus taskStatusArray[PROFILE_MOVE_TASK_INDEX + 1];
+    Automation1TaskStatus* taskStatus;
+    int executeState;
+    bool pollOk = true;
+
+    if (!dataCollectionConfig_) goto done;
+
+    getIntegerParam(profileExecuteState_, &executeState);
+    if (executeState == PROFILE_EXECUTE_EXECUTING)
+    {
+        if (!Automation1_DataCollection_GetStatus(controller_, &dataCollectionStatus))
+        {
+            logApiError(profileExecuteMessage_, "Failed to get data collection status");
+            pollOk = false;
+            goto done;
+        }
+
+        setIntegerParam(profileCurrentPoint_, dataCollectionStatus.NumberOfRetrievedPoints);
+
+        if (!dataCollectionStatus.IsCollecting) {
```

Comment:
Instead of checking if data collection is still collecting, it might be more correct to check the task state of **PROFILE_MOVE_TASK_INDEX** via `Automation1TaskStatus.TaskState == Automation1TaskState_ProgramRunning`.

Or perhaps checking both task state *and* data collection status.